### ISSUE TITLE
Unable to perform a count operation on a named query

### ISSUE THREAD

**Issue by jkalex-git** [quarkusio/quarkus#20758]
### Describe the bug

Panache count not working with named queries.

```
import io.quarkus.hibernate.orm.panache.PanacheEntity;
import javax.persistence.*;

@Entity
@Cacheable
@NamedQueries({
        @NamedQuery(name = "countFruit",
                query = "FROM Fruit")
})
public class Fruit extends PanacheEntity {

    @Column(length = 40, unique = true)
    public String name;


    public static long countFruits() {
        return count("#countFruit");   // This is not working
    }


    public static long countFruitsByFind() {
        return find("#countFruit").count();     // This is working
    }
}
```

### Expected behavior

The panache count query works with named queries.

### Actual behavior

**Stacktrace:**

```
2021-10-14 11:00:23,948 ERROR [org.acm.hib.orm.pan.FruitResource] (executor-thread-0) Failed to handle request: java.lang.IllegalArgumentException: org.hibernate.QueryException: unexpected char: '#' [SELECT COUNT(*) FROM org.acme.hibernate.orm.panache.Fruit WHERE #countFruit]
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:138)
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:181)
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:188)
	at org.hibernate.internal.AbstractSharedSessionContract.createQuery(AbstractSharedSessionContract.java:743)
	at org.hibernate.internal.AbstractSharedSessionContract.createQuery(AbstractSharedSessionContract.java:114)
	at io.quarkus.hibernate.orm.runtime.session.TransactionScopedSession.createQuery(TransactionScopedSession.java:345)
	at io.quarkus.hibernate.orm.runtime.session.ForwardingSession.createQuery(ForwardingSession.java:168)
	at io.quarkus.hibernate.orm.runtime.session.ForwardingSession.createQuery(ForwardingSession.java:47)
	at org.hibernate.Session_5b93bee577ae2f8d76647de04cfab36afbf52958_Synthetic_ClientProxy.createQuery(Session_5b93bee577ae2f8d76647de04cfab36afbf52958_Synthetic_ClientProxy.zig:1916)
	at io.quarkus.hibernate.orm.panache.common.runtime.AbstractJpaOperations.count(AbstractJpaOperations.java:317)
	at org.acme.hibernate.orm.panache.Fruit.count(Fruit.java)
	at org.acme.hibernate.orm.panache.Fruit.countFruits(Fruit.java:19)
	at org.acme.hibernate.orm.panache.FruitResource.getCount(FruitResource.java:54)
	at org.acme.hibernate.orm.panache.FruitResource_Subclass.getCount$$superforward1(FruitResource_Subclass.zig:378)
	at org.acme.hibernate.orm.panache.FruitResource_Subclass$$function$$4.apply(FruitResource_Subclass$$function$$4.zig:24)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.proceed(InvocationInterceptor.java:62)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.monitor(InvocationInterceptor.java:49)
	at io.quarkus.arc.runtime.devconsole.InvocationInterceptor_Bean.intercept(InvocationInterceptor_Bean.zig:521)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
	at org.acme.hibernate.orm.panache.FruitResource_Subclass.getCount(FruitResource_Subclass.zig:748)
	at org.acme.hibernate.orm.panache.FruitResource_ClientProxy.getCount(FruitResource_ClientProxy.zig:191)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:408)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:69)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:135)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:90)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:543)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.hibernate.QueryException: unexpected char: '#' [SELECT COUNT(*) FROM org.acme.hibernate.orm.panache.Fruit WHERE #countFruit]
	at org.hibernate.hql.internal.ast.QueryTranslatorImpl.doCompile(QueryTranslatorImpl.java:236)
	at org.hibernate.hql.internal.ast.QueryTranslatorImpl.compile(QueryTranslatorImpl.java:144)
	at org.hibernate.engine.query.spi.HQLQueryPlan.<init>(HQLQueryPlan.java:113)
	at org.hibernate.engine.query.spi.HQLQueryPlan.<init>(HQLQueryPlan.java:73)
	at org.hibernate.engine.query.spi.QueryPlanCache.getHQLQueryPlan(QueryPlanCache.java:162)
	at org.hibernate.internal.AbstractSharedSessionContract.getQueryPlan(AbstractSharedSessionContract.java:622)
	at org.hibernate.internal.AbstractSharedSessionContract.createQuery(AbstractSharedSessionContract.java:734)
	... 50 more
```

### Output of `java -version`

openjdk version "11.0.11" 

### Quarkus version or git rev

2.3.0.FINAL

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.6.3

**Comment by infeo** [comment]
Also `PanacheEntitybase.delete(...)` does not work with named queries, same cause.  Maybe even more methods.

**Comment by loicmathieu** [comment]
I updated my previous comment, so named queries are only supported for querying.

It should also be supported for count(String)/delete(String)/update(String)

**Comment by enwooz** [comment]
Thx, I will try to reproduce issue in unit test, and after if it's confirmed, I'm going to fix it.

**Comment by gilvansfilho** [comment]
Hi Guys, I believe I found the error and get a solution to that.

If I am right the problem is in `count` methods from `io.quarkus.hibernate.orm.panache.common.runtime.AbstractJpaOperations`
They dont check if query is a named query handling everything as not named then forwading query to `io.quarkus.panache.hibernate.common.runtime.PanacheJpaUtil.createCountQuery` method. As neither condition is true the method return this

`return "SELECT COUNT(*) FROM " + getEntityName(entityClass) + " WHERE " + query;`

getting the error exposed by @jkalex-git 

`2021-10-14 11:00:23,948 ERROR [org.acm.hib.orm.pan.FruitResource] (executor-thread-0) Failed to handle request: java.lang.IllegalArgumentException: org.hibernate.QueryException: unexpected char: '#' [SELECT COUNT(*) FROM org.acme.hibernate.orm.panache.Fruit WHERE #countFruit]`

I think that a simple solution is check, in `io.quarkus.hibernate.orm.panache.common.runtime.JpaOperations`, if query is a named query and if so do this:

```
String namedQuery = query.substring(1);
NamedQueryUtil.checkNamedQuery(entityClass, namedQuery);
return createPanacheQuery(em, query, PanacheJpaUtil.toOrderBy(sort), params).count();
```

If not forward to super `io.quarkus.hibernate.orm.panache.common.runtime.AbstractJpaOperations` to handle as is.

I already [did it](https://github.com/gilvansfilho/quarkus/commit/188cb3845aae37f0b6f457e35df58a19c22a3b62) in my repo and works fine. 

Could I submit a PR?

**Comment by loicmathieu** [comment]
Fixed by #21023
